ingestion_program/datasets.py

```diff
@@ -73,31 +78,79 @@ def load_train_set(self):
             self.input_dir, "train", "detailed_labels", "data.detailed_labels"
         )
 
-        # read train labels
-        with open(train_labels_file, "r") as f:
-            train_labels = np.array(f.read().splitlines(), dtype=float)
-
-        # read train settings
-        with open(train_settings_file) as f:
-            train_settings = json.load(f)
+        parquet_file = pq.ParquetFile(train_data_file)
+
+        # Step 1: Determine the total number of rows
+        total_rows = sum(parquet_file.metadata.row_group(i).num_rows for i in range(parquet_file.num_row_groups))
+
+        if sample_size is not None:
+            if isinstance(sample_size, int):
+                sample_size = min(sample_size, total_rows)
+            elif isinstance(sample_size, float):
+                if 0.0 <= sample_size <= 1.0:
+                    sample_size = int(sample_size * total_rows)
+                else :
+                    raise ValueError("Sample size must be between 0.0 and 1.0")
+            else:
+                raise ValueError("Sample size must be an integer or a float")
+        elif selected_indices is not None:
+            sample_size = len(selected_indices)
+        else:
+            sample_size = total_rows
+ 
+        if selected_indices is None:
+            selected_indices = np.sort(np.random.choice(total_rows, size=sample_size, replace=False))
+        
+        selected_indices_set = set(selected_indices)
+
+
+        def get_sampled_data(data_file):
+            selected_list = []
+            with open(data_file, "r") as f:
+                for i, line in enumerate(f):
+                    # Check if the current line index is in the selected indices
+                    if i not in selected_indices_set:
+                        continue
+                    if data_file.endswith(".detailed_labels"):
+                        selected_list.append(line.strip())
+                    else:
+                        selected_list.append(float(line.strip()))
+                    # Optional: stop early if all indices are found
+                    if len(selected_list) == len(selected_indices):
+                        break
+            return np.array(selected_list)
+
+        current_row = 0
+        sampled_df = pd.DataFrame()
+        for row_group_index in range(parquet_file.num_row_groups):
+            row_group = parquet_file.read_row_group(row_group_index).to_pandas()
+            row_group_size = len(row_group)
+            
+            # Determine indices within the current row group that fall in the selected range
+            within_group_indices = selected_indices[(selected_indices >= current_row) & (selected_indices < current_row + row_group_size)] - current_row
```

Comment:
This condition gives you an error when selected_indices is used. You can test it with this code

```python
from datasets import Data

data = Data("../input_data/")
data.load_train_set(selected_indices=[1,4,5,6])
```

Error
```
within_group_indices = selected_indices[(selected_indices >= current_row) & (selected_indices < current_row + row_group_size)] - current_row
TypeError: '>=' not supported between instances of 'list' and 'int'
```

Reply:
with np array I don't get any error

```python
data.load_train_set(selected_indices=np.array([1,4,5,6]))
```